pkg/informers/context.go

```diff
@@ -0,0 +1,51 @@
+package informers
+
+import (
+	"context"
+
+	"k8s.io/client-go/informers"
+
+	argorolloutsinformer "github.com/datawire/argo-rollouts-go-client/pkg/client/informers/externalversions"
+	"github.com/datawire/k8sapi/pkg/k8sapi"
+)
+
+type factoryKey string
+
+func getOpts(ns string) (k8sOpts []informers.SharedInformerOption, argoOpts []argorolloutsinformer.SharedInformerOption) {
+	if ns != "" {
+		k8sOpts = append(k8sOpts, informers.WithNamespace(ns))
+		argoOpts = append(argoOpts, argorolloutsinformer.WithNamespace(ns))
+	}
+
+	return k8sOpts, argoOpts
+}
+
+func WithInformers(ctx context.Context, ns string) context.Context {
```

Comment:
I'm not happy with the naming here, because it introduces stuttering and provides less information. In general, I think we should be careful using plural in package names.

Consider `informers.WithInformers` compared to `informer.WithFactory` or `informers.GetInformers` compared to `informer.GetFactory`

Perhaps rename the `GlobalInformers` to `Factory`? That will result in `informer.Factory` when used, which IMO provides more hints than `informers.GlobalInformers`. `Global` is actually not true, because in a namespaced setting, the informer will only look at one namespace.

Reply:
I agree, I didn't like that name either, but couldn't think of a better name